services/gundeck/test/integration/API.hs

```diff
@@ -614,33 +614,19 @@ testRegisterPushToken g _ b _ = do
 
 -- TODO: Try to make this test more performant, this test takes too long right now
 testRegisterTooManyTokens :: TestSignature ()
-testRegisterTooManyTokens g _ b _ = do
+testRegisterTooManyTokens g _ _ _ = do
     -- create tokens for reuse with multiple users
-    apsTok <- Token . T.decodeUtf8 . B16.encode <$> randomBytes 32
     gcmTok <- Token . T.decodeUtf8 . toByteString' <$> randomId
-    
+    uids   <- liftIO $ replicateConcurrently 55 randomId
     -- create 55 users with these tokens, which should succeed
-    userClients <- replicateM 55 $ registerToken 201 apsTok gcmTok
-
+    mapM_ (registerToken 201 gcmTok) uids
```

Comment:
Can't these be done in parallel?

Reply:
Unfortunately these cannot, as concurrent updates would break the test.